modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java

```diff
@@ -525,11 +576,11 @@ private SegmentsExperiment _updateWinnerSegmentsExperienceId(
 			(winnerSegmentsExperienceId !=
 				segmentsExperiment.getSegmentsExperienceId())) {
 
-			_segmentsExperienceLocalService.updateSegmentsExperienceActive(
-				segmentsExperiment.getSegmentsExperienceId(), false);
-
-			_segmentsExperienceLocalService.updateSegmentsExperienceActive(
-				winnerSegmentsExperienceId, true);
+			_publishSegmentsExperienceVariant(
```

Comment:
Should this method return the segmentExperience that is updating?

Reply:
good question, I had another name before, but, with this one, maybe It's better to return the published variant. Thanks!